### PR TITLE
Batch keys specified by user and specify batch processing function ra…

### PR DESCRIPTION
**Pull request by samedii**
…ther than create trainer

Uses decorators to make it easier to create a custom `process_batch` function.

Related to #5 #2

Also added "recipes" directory for common usage examples etc. that do not fit in as a function (yet).

@FelixAbrahamsson @Jim-Holmstroem What do you think? I think this should require two reviews / upvotes to merge. Also feel free to comment the code and let's adjust it together

### CODE REVIEW

**Comment by samedii** [comment]
Usage in coop:


    @workflow.ignite.decorators.train(
        model, optimizer, batches_per_step=1
    )
    def train_batch(engine, batch):
        batch['predictions'] = model(batch['features'])
        loss_ = F.binary_cross_entropy_with_logits(
            batch['predictions'], batch['targets']
        )
        loss_.backward()
        batch['loss'] = loss_.item()
        return batch

    trainer = ignite.engine.Engine(train_batch)

    workflow.ignite.handlers.attach_train_progress_bar(trainer, dict(
        max_epochs=max_epochs, batches_per_epoch=batches_per_epoch
    ))

    trainer.run(map(lambda x: x, train_data_iterator), max_epochs=max_epochs, epoch_length=batches_per_epoch)

**Comment by samedii** [comment]
1b7dcf0 Fixes bug in step that @FelixAbrahamsson found.
Edit: Too enthusiastic with committing - works now.

File: workflow/ignite/decorators/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+from workflow.ignite.decorators import (
+    to_device, no_grad
+)
+
+
+def evaluate(model):
+    device = model_device(model)
+
+    def decorator(process_batch):
+
```
Comment (Jim-Holmstroem) [inline workflow/ignite/decorators/evaluate.py]:
```suggestion
@wraps(process_batch)
```

**Comment by Jim-Holmstroem** [comment]
sorry, mannen trodde inte github skulle commita den :P

**Review by FelixAbrahamsson — APPROVED** [review]
